Add LIBJXL_FORCE_SYSTEM_LCMS

afaict no special patches are needed for lcms in libjxl, so it should be ok to use the system version. right now i am using https://cgit.alxu.ca/gentoo-overlay.git/tree/media-libs/libjxl/files/system-lcms.patch, but I'm not sure if `target_link_libraries(lcms2 INTERFACE PkgConfig::LCMS)` works on cmake 3.10, also i think my interpretation of FORCE_SYSTEM was not consistent with the other usages.

Using the patch above, I kept getting an error:

```
CMake Error at third_party/CMakeLists.txt:207 (add_library):
  INTERFACE_LIBRARY targets may only have whitelisted properties.  The
  property "INCLUDE_DIRECTORIES" is not allowed.
```

You need to fix (lib/jxl.cmake):

```
  target_include_directories(jxl_enc-obj PRIVATE
    $<TARGET_PROPERTY:lcms2,INCLUDE_DIRECTORIES>
  )
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -122,10 +122,12 @@ set(JPEGXL_FORCE_NEON false CACHE BOOL
 
 
 # Force system dependencies.
-set(JPEGXL_FORCE_SYSTEM_GTEST false CACHE BOOL
-    "Force using system installed googletest (gtest/gmock) instead of third_party/googletest source.")
 set(JPEGXL_FORCE_SYSTEM_BROTLI false CACHE BOOL
     "Force using system installed brotli instead of third_party/brotli source.")
+set(JPEGXL_FORCE_SYSTEM_GTEST false CACHE BOOL
+    "Force using system installed googletest (gtest/gmock) instead of third_party/googletest source.")
+set(JPEGXL_FORCE_SYSTEM_LCMS2 false CACHE BOOL
+    "Force using system installed lcms2 instead of third_party/lcms source.")
 set(JPEGXL_FORCE_SYSTEM_HWY false CACHE BOOL
     "Force using system installed highway (libhwy-dev) instead of third_party/highway source.")
 

--- a/lib/jxl.cmake
+++ b/lib/jxl.cmake
@@ -402,13 +402,9 @@ endif()
 
 #TODO(lode): don't depend on CMS for the core library
 if (JPEGXL_ENABLE_SKCMS)
-  target_include_directories(jxl_enc-obj PRIVATE
-    $<TARGET_PROPERTY:skcms,INCLUDE_DIRECTORIES>
-  )
+  target_link_libraries(jxl_enc-obj PRIVATE skcms)
 else ()
-  target_include_directories(jxl_enc-obj PRIVATE
-    $<TARGET_PROPERTY:lcms2,INCLUDE_DIRECTORIES>
-  )
+  target_link_libraries(jxl_enc-obj PRIVATE lcms2)
 endif ()
 
 # Headers for exporting/importing public headers

